### PR TITLE
Support for both Cookie libraries for different versions of Spring

### LINKED ISSUE #2542: `ClassNotFoundException: javax.servlet.http.Cookie` when generating controller tests
**Description**

Generating controller tests fails due to `ClassNotFoundException: javax.servlet.http.Cookie`.

**To Reproduce**

Generate integration tests for `OwnerController.processFindForm()` from [`spring-petclinic` project](https://github.com/spring-projects/spring-petclinic) using `PetClinicApplication` configuration.

**Expected behavior**

Tests are generated without errors.

**Actual behavior**

No tests are generated, there's the following error in the `utbot-engine-current.log`

```
13:01:10.801 | INFO  | JavaLanguage              | Fuzzing is stopped because of an error
java.lang.ClassNotFoundException: javax.servlet.http.Cookie
	at java.net.URLClassLoader.findClass(URLClassLoader.java:445) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:587) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:520) ~[?:?]
	at org.utbot.framework.plugin.api.util.SpringModelUtils.createRequestBuilderModelOrNull(SpringModelUtils.kt:328) ~[utbot-framework-api-2023.08-SNAPSHOT.jar:?]
	at org.utbot.framework.context.spring.SpringIntegrationTestJavaFuzzingContext.createStateBefore(SpringIntegrationTestJavaFuzzingContext.kt:97) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.engine.UtBotSymbolicEngine$fuzzing$1$1.invokeSuspend(UtBotSymbolicEngine.kt:482) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.engine.UtBotSymbolicEngine$fuzzing$1$1.invoke(UtBotSymbolicEngine.kt) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.engine.UtBotSymbolicEngine$fuzzing$1$1.invoke(UtBotSymbolicEngine.kt) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.fuzzing.JavaLanguageKt$runJavaFuzzing$5.invokeSuspend(JavaLanguage.kt:131) ~[utbot-java-fuzzing-2023.08-SNAPSHOT.jar:?]
	at org.utbot.fuzzing.JavaLanguageKt$runJavaFuzzing$5.invoke(JavaLanguage.kt) ~[utbot-java-fuzzing-2023.08-SNAPSHOT.jar:?]
	at org.utbot.fuzzing.JavaLanguageKt$runJavaFuzzing$5.invoke(JavaLanguage.kt) ~[utbot-java-fuzzing-2023.08-SNAPSHOT.jar:?]
	at org.utbot.fuzzing.BaseFuzzing.handle(Providers.kt:56) ~[utbot-fuzzing-2023.08-SNAPSHOT.jar:?]
	at org.utbot.fuzzing.FuzzingApi.fuzz(Api.kt:359) ~[utbot-fuzzing-2023.08-SNAPSHOT.jar:?]
	at org.utbot.fuzzing.FuzzingApi.access$fuzz(Api.kt:1) ~[utbot-fuzzing-2023.08-SNAPSHOT.jar:?]
	at org.utbot.fuzzing.FuzzingApi$fuzz$2.invokeSuspend(Api.kt) ~[utbot-fuzzing-2023.08-SNAPSHOT.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) ~[kotlin-stdlib-1.8.10.jar:1.8.10-release-430(1.8.10)]
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:284) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source) ~[kotlinx-coroutines-core-jvm-1.6.3.jar:?]
	at org.utbot.common.ConcurrencyKt.runBlockingWithCancellationPredicate(Concurrency.kt:38) ~[utbot-core-2023.08-SNAPSHOT.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:196) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator$generate$2$3.invoke(TestCaseGenerator.kt:195) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.common.ConcurrencyKt.runIgnoringCancellationException(Concurrency.kt:47) ~[utbot-core-2023.08-SNAPSHOT.jar:?]
	at org.utbot.framework.plugin.api.TestCaseGenerator.generate(TestCaseGenerator.kt:195) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.framework.process.EngineProcessMainKt$setup$5.invoke(EngineProcessMain.kt:145) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.framework.process.EngineProcessMainKt$setup$5.invoke(EngineProcessMain.kt:130) ~[utbot-framework-2023.08-SNAPSHOT.jar:?]
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1$2$1.invoke(ClientProcessUtil.kt:115) ~[utbot-rd-2023.08-SNAPSHOT.jar:?]
	at org.utbot.rd.IdleWatchdog.wrapActive(ClientProcessUtil.kt:88) ~[utbot-rd-2023.08-SNAPSHOT.jar:?]
	at org.utbot.rd.IdleWatchdog$measureTimeForActiveCall$1.invoke(ClientProcessUtil.kt:114) ~[utbot-rd-2023.08-SNAPSHOT.jar:?]
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.IRdEndpoint$set$1.invoke(TaskInterfaces.kt:182) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.RdCall.onWireReceived(RdTask.kt:362) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:57) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.impl.ProtocolContexts.readMessageContextAndInvoke(ProtocolContexts.kt:148) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:56) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.framework.MessageBroker$invoke$2.invoke(MessageBroker.kt:54) ~[rd-framework-2023.1.2.jar:?]
	at com.jetbrains.rd.util.threading.SingleThreadSchedulerBase.queue$lambda-3(SingleThreadScheduler.kt:41) ~[rd-core-2023.1.2.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
	at java.lang.Thread.run(Thread.java:833) [?:?]
```

**Additional context**

`javax.servlet.http.Cookie` is replaced with `jakarta.servlet.http.Cookie` in newer versions of Spring, we need to support both.

### PR DESCRIPTION
**Pull request by tepa46**
## Description

Fixes #2542

Added support for `jakarta.servlet.http.Cookie` which is used in newer versions of Spring.

### Manual tests
Tested on the `spring-petclinic` project. (Spring Boot 3.1.3 with `jakarta.servlet.http.Cookie`)
Tested on the `spring-boot-testing` project. (Spring Boot 2.7.3 with `javax.servlet.http.Cookie`)

## Self-check list

- [x] I've set the proper **labels** for my PR (at least, for category and component).
- [x] PR **title** and **description** are clear and intelligible.
- [x] I've added enough **comments** to my code, particularly in hard-to-understand areas.
- [ ] The functionality I've repaired, changed or added is covered with **automated tests**.
- [x] **Manual tests** have been provided optionally.
- [x] The **documentation** for the functionality I've been working on is up-to-date.

### CODE REVIEW

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt [hunk 1]
```diff
@@ -643,6 +653,7 @@ object SpringModelUtils {
     }
 
     private fun createCookieValuesModel(
+        cookieClassId: ClassId,
```
Comment (EgorkaKulikov) [inline utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt:656]: I do not  catch an idea here. What for do we pass an argument and do not use it?

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt [hunk 2]
```diff
@@ -413,10 +418,15 @@ object SpringModelUtils {
         val headersContentModel = createHeadersContentModel(methodId, arguments, idGenerator)
         requestBuilderModel = addHeadersToRequestBuilderModel(headersContentModel, requestBuilderModel, idGenerator)
 
-//      // TODO uncomment when #2542 is fixed
-//        val cookieValuesModel = createCookieValuesModel(methodId, arguments, idGenerator)
-//        requestBuilderModel =
-//            addCookiesToRequestBuilderModel(cookieValuesModel, requestBuilderModel, idGenerator)
+        val cookieClassId = cookieClassIds.singleOrNull()
```
Comment (EgorkaKulikov) [inline utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt:421]: Why not firstOrNull? Is it guarantied that a user may not use them in common (even without any real motivation for it)?

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt [hunk 3]
```diff
@@ -413,10 +418,15 @@ object SpringModelUtils {
...
+        if(cookieClassId is ClassId) {
```
Comment (EgorkaKulikov) [inline utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt:422]: This is an equivalent of cookieClassIs is null? Why not `?.let { ... } ?: warn { ...}` then? Anyway, the semantic of your check is not clear.

**Review by EgorkaKulikov — CHANGES_REQUESTED** [review]

File: utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt (hunk 2 above)
Comment (tepa46) [reply utbot-framework-api/src/main/kotlin/org/utbot/framework/plugin/api/util/SpringModelUtils.kt:421]: Depending on which version of Spring is used in the user project, Spring uses different cookie libraries. If the user uses them in common, it is unclear which of the libraries use to build models.

**Review by EgorkaKulikov — APPROVED** [review]

**Pull request merged**